### ISSUE TITLE
CheckComboBox - at least one item selected

### ISSUE THREAD

**Issue by FroggieFrog** [xceedsoftware/wpftoolkit#1507]
Hi!
I'm using the free edition of the `CheckComboBox` control. Thank you for your work!

I'm trying to create a `CheckComboBox` with an "All" option. the `CheckComboBox` is bound via MVVM to an `ObservableList`.
Now when the "All" option is selected all other options should be deselected.
But when the "All" option is selected and the user clicks the "All" option then nothing should happen, because there needs to be at least one option selected.

How can I achieve this?

**Comment by FroggieFrog** [comment]
Maybe a little code makes it easier to understand what I want.
I don't want the user to be able to deselect "All" if it is the only selected value.

I tried different things inside of `EntryClicked`, but I was not able to find a working solution.

```csharp
    public class ViewModel : NotifyBase
    {
        private ObservableCollection<Model> _availableModels;
        private ObservableCollection<Model> _selectedModels;
        private bool _isUpdating;

        public ViewModel()
        {
            EntryClickedCommand = new RelayCommand(EntryClicked);

            AvailableModels = new ObservableCollection<Model>();
            SelectedModels = new ObservableCollection<Model>();
            InitAvailableModels();
            SelectedModels.Add(AvailableModels.Single(x => x.Type == MyEnum.All));
        }

        private void InitAvailableModels()
        {
            //add "All" at first position
            AvailableModels.Add(new Model
            {
                Name = MyEnum.All.ToString(),
                Type = MyEnum.All,
                IsSelected = true
            });

            foreach (MyEnum curEnumValue in Enum.GetValues(typeof(MyEnum)))
            {
                if (curEnumValue == MyEnum.Unknown || curEnumValue == MyEnum.All)
                {
                    continue;
                }

                AvailableModels.Add(new Model
                {
                    Name = curEnumValue.ToString(),
                    Type = curEnumValue
                });
            }
        }

        public ObservableCollection<Model> AvailableModels
        {
            get { return _availableModels; }
            set { SetPropertyAndNotify(ref _availableModels, value); }
        }

        public ObservableCollection<Model> SelectedModels
        {
            get { return _selectedModels; }
            set { SetPropertyAndNotify(ref _selectedModels, value); }
        }

        public ICommand EntryClickedCommand { get; private set; }

        private void EntryClicked(object param)
        {
            if (_isUpdating)
            {
                return;
            }

            try
            {
                _isUpdating = true;

                if (!(param is Model clickedEntry))
                {
                    return;
                }

                var allEntry = AvailableModels.Single(x => x.Type == MyEnum.All);

                //this triggers an endless loop -> crash
                ////at least "All" has to be selected
                //if (SelectedModels.Count == 0 && clickedEntry.Type == MyEnum.All)
                //{
                //    //allEntry.IsSelected = true;
                //    //SelectedModels.Clear();
                //    SelectedModels.Add(AvailableModels.Single(x => x.Type == MyEnum.All));
                //    return;
                //}

                if (clickedEntry.Type != MyEnum.All)
                {
                    allEntry.IsSelected = false;
                    clickedEntry.IsSelected = true;
                }
                else
                {
                    foreach (var curEntry in AvailableModels.Where(x => x.Type != MyEnum.All))
                    {
                        curEntry.IsSelected = false;
                    }

                    allEntry.IsSelected = true;
                }
            }
            finally
            {
                _isUpdating = false;

                OnPropertyChanged(nameof(SelectedModels));
            }
        }
    }

    public class Model : NotifyBase
    {
        private MyEnum _myEnum;
        private string _name;
        private bool _isSelected;

        public MyEnum Type
        {
            get { return _myEnum; }
            set { SetPropertyAndNotify(ref _myEnum, value); }
        }

        public string Name
        {
            get { return _name; }
            set { SetPropertyAndNotify(ref _name, value); }
        }

        public bool IsSelected
        {
            get { return _isSelected; }
            set { SetPropertyAndNotify(ref _isSelected, value); }
        }

    }

    [Flags]
    public enum MyEnum
    {
        Unknown = 0,
        Foo = 1 << 0,
        Bar = 1 << 1,
        Baz = 1 << 2,
        All = ~Unknown
    }

    public class NotifyBase : INotifyPropertyChanged
    {
        public event PropertyChangedEventHandler PropertyChanged;

        protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
        {
            PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
        }

        protected virtual bool SetPropertyAndNotify<T>(ref T backingField, T value, [CallerMemberName] string propertyName = "")
        {
            if (EqualityComparer<T>.Default.Equals(backingField, value))
            {
                return false;
            }

            backingField = value;
            OnPropertyChanged(propertyName);
            return true;
        }
    }
```
 
```xaml
<xctk:CheckComboBox Grid.Row="0"
                            IsEditable="False"
                            ItemsSource="{Binding AvailableModels}"
                            DisplayMemberPath="Name"
                            SelectedMemberPath="IsSelected"
                            Delimiter=", "
                            Command="{Binding EntryClickedCommand}"
                            SelectedItemsOverride="{Binding SelectedModels}" />
```

**Comment by XceedBoucherS** [comment]
Hi,
Before testing your code, have you try the IsSelectAllActive property on the CheckComboBox ? This feature was introduced in v3.4
When set to true, it will add a "Select All" option in the checkComboBox popup. Clicking this option will check all the other options and unselecting it will unselect all the other options. Also, when some of the other options are selected, the "Select All" option will show a black square, meaning its between 2 states (not all the other options are selected or unselected). Also, if you select(or unselect) all the other options, the "Select All" option will be correctly updated.

**Comment by FroggieFrog** [comment]
Thank you for your reply.
I didn't know this feature because the [wiki page](https://github.com/xceedsoftware/wpftoolkit/wiki/CheckComboBox) is missing it.
I will try it and let you know if it works as expected.
Is there support for localizing the text "Select All"?

**Comment by FroggieFrog** [comment]
I tested your solution with the `IsSelectAllActive` property and it worked as intended.
Also (by accident) I found the property `SelectAllContent` to use for translation.

@XceedBoucherS
Sadly the wiki/documentation is missing both properties. So if you know a way I can contribute I would be glad to submit a PR for the wiki.

I close this issue, because I have a solution.
Thank you again for your help!
